source/utilities.cc

```diff
@@ -2277,6 +2277,182 @@ namespace aspect
+                                                        subsection_name);
+
+      prm.enter_subsection (subsection_name);
+      {
+        prm.declare_entry ("Data directory",
+                           default_directory,
+                           Patterns::DirectoryName (),
+                           "The name of a directory that contains the model data. This path "
+                           "may either be absolute (if starting with a `/') or relative to "
+                           "the current directory. The path may also include the special "
+                           "text `$ASPECT_SOURCE_DIR' which will be interpreted as the path "
+                           "in which the ASPECT source files were located when ASPECT was "
+                           "compiled. This interpretation allows, for example, to reference "
+                           "files located in the `data/' subdirectory of ASPECT. ");
+        prm.declare_entry ("Data file names",
+                           default_filename,
+                           Patterns::List (Patterns::Anything()),
+                           "The file names of the model data (comma separated). ");
+
+        prm.declare_entry ("Layer boundary values", "1200",
+                           Patterns::List (Patterns::Anything()),
+                           "The list of layer values to be read in. These must be given "
+                           "in ascending order of the vertical component inside the data file. "
+                           "This is not necessarily in ascending order of the boundary values. "
+                           "The boundary values should correspond to the ascii file data names.");
+
+        prm.declare_entry ("Interpolation scheme", "linear",
+                           Patterns::Selection("piecewise constant|linear"),
+                           "Method to interpolate between layers. Select from "
+                           "piecewise constant or linear. Piecewise constant takes the "
+                           "value from the nearest layer boundary above the data point."
+                           "Above and below the domain given by the layers, the values are"
+                           "given by the top and bottom layer boundary.");
+
+      }
+      prm.leave_subsection();
+    }
+
+
+    template <int dim>
+    void
+    AsciiDataLayered<dim>::parse_parameters (ParameterHandler &prm,
+                                             const std::string &subsection_name)
+    {
+      Utilities::AsciiDataBase<dim>::parse_parameters(prm, subsection_name);
+
+      prm.enter_subsection(subsection_name);
+      {
+
+        data_directory = Utilities::expand_ASPECT_SOURCE_DIR(prm.get ("Data directory"));
+        data_file_names    = Utilities::split_string_list(prm.get ("Data file names"), ',');
+        layer_boundary_names = Utilities::split_string_list(prm.get("Layer boundary values"), ',');
```

Comment:
are those names or values?!

Reply:
Ooops, I used this parameter when I had a different file-finding structure. Changed!